core/src/main/java/org/testcontainers/containers/DockerComposeContainer.java

```diff
@@ -429,6 +436,19 @@ public ContainerisedDockerCompose(List<File> composeFiles, String identifier) {
         addEnv("DOCKER_HOST", "unix:///docker.sock");
         setStartupCheckStrategy(new IndefiniteWaitOneShotStartupCheckStrategy());
         setWorkingDirectory(containerPwd);
+
+        String dockerConfigEnv = System.getenv(DOCKER_CONFIG_ENV);
+        String dockerConfigProperty = System.getProperty(DOCKER_CONFIG_PROPERTY);
+        Path dockerConfig = Paths.get(System.getProperty("user.home"), ".docker", "config.json");
+
+        if (dockerConfigEnv != null && !dockerConfigEnv.trim().isEmpty() && Files.exists(Paths.get(dockerConfigEnv))) {
```

Comment:
Could you please encapsulate this boolean expressions into private methods with a speaking name? Like `dockerConfigEnvExists`, `dockerConfigPropertyExists`, etc.

Reply:
It looks like this is the last thing still to do on this PR - is that correct? I can do this quickly tomorrow if you're stuck for time @yurybubnov. It'd be no problem.
Thanks for this PR!

Reply:
I moved it to separate method `determineDockerConfigPath `
https://github.com/testcontainers/testcontainers-java/pull/537/files#diff-f992b47914d235d916c506932b1946f6R446